cocotb/utils.py

```diff
@@ -35,12 +35,7 @@
 import functools
 import warnings
 
-if "COCOTB_SIM" in os.environ:
-    from cocotb import simulator
-    _LOG_SIM_PRECISION = simulator.get_precision()  # request once and cache
-else:
-    simulator = None
-    _LOG_SIM_PRECISION = -15
```

Comment:
Seems a shame we can't cache this - do we have a feel for how expensive `simulator.get_precision()` is on some common simulators? Going from `Python -> C -> GPI -> VPI` seems like a fair bit of overhead, but perhaps its not.

One option would be to use my IPython branch and use `%timeit simulator.get_precision()`

Reply:
I'll test the timing, but we could just cache the value in the simulator module, it cannot change during the run.

Reply:
Or we could cache it in a local function.

Reply:
timeit reports `simulator.get_precision()` takes 112ns on my laptop, and the cached local function impl takes 78ns.

Reply:
Caching it in the `simulator.get_precision` might be sensible too

Reply:
Just tested that at 103ns.